### ISSUE TITLE
GETTEXT Installation issue

### ISSUE THREAD

**Issue by anthonyjwall** [StefanSchippers/xschem-gaw#11]
Hi! Thank you for the tool! I've been having some issues installing, different to the error with the GETTEXT version you online in the main README.md

# Issue
Running `./configure` yields an error:
``./configure: line 5347: syntax error near unexpected token `external'``
``./configure: line 5347: `AM_GNU_GETTEXT(external)'``

# System
I've tried to install the repo this on Ubuntu 20.04 and Debian 10, with the same result. 
I've got gettext version 0.19.8.1 installed.

# Steps to reproduce
## Clone the git repo
``$ git clone git@github.com:stefanschippers/xschem-gaw``

## Modify ``Makefile.in.in``
``$cd xschem-gaw/``
``$ vim po/Makefile.in.in``
``GETTEXT_MACRO_VERSION = 0.19``

Running ``aclocal``: 
``$ aclocal``
``configure.ac:70: warning: macro 'AM_GNU_GETTEXT' not found in library``
``configure.ac:71: warning: macro 'AM_GNU_GETTEXT_VERSION' not found in library``

Running ``automake``:
``$ automake --add-missing``
``src/Makefile.am:19: warning: 'INCLUDES' is the old name for 'AM_CPPFLAGS' (or '*_CPPFLAGS')``

Running ``autoconf``:
``$autoconf``
``configure.ac:73: error: possibly undefined macro: AM_GNU_GETTEXT``
``      If this token and others are legitimate, please use m4_pattern_allow.``
``      See the Autoconf documentation.``
``configure.ac:74: error: possibly undefined macro: AM_GNU_GETTEXT_VERSION``

Running ``./configure``:
``./configure``
etc...etc...
``checking for sys/endian.h... no``
``./configure: line 5347: syntax error near unexpected token `external'``
``./configure: line 5347: `AM_GNU_GETTEXT(external)'``

I know you've had other issues with the version, and you're not an expert in autotools - but have you any idea where I could start digging to sort this issue? Apologies if this is too vague or I'm doing something silly.

Thanks!

**Comment by StefanSchippers** [comment]
Is gettext installed on the system?
Does the gettext version match the one installed on your system?
The GETTEXT_MACRO_VERSION should match the gettext verison on the system.
to check: 

`dpkg -l | grep gettext`

**Comment by StefanSchippers** [comment]
On my system i have gettext, autopoint, autoconf, automake installed.
If some of these are missing try to install and retry.

**Comment by anthonyjwall** [comment]
Hi Stefan, I've got gettext installed on the system, and I've set GETTEXT_MACRO_VERSION=0.19 to match the version (0.19.8.1):
``dpkg -l | grep gettext``
``ii  gettext-base                                  0.19.8.1-10build1                           amd64        GNU Internationalization utilities for the base system``
``ii  liblocale-gettext-perl                        1.07-4                                      amd64        module using libc functions for internationalization in Perl``


I've made sure gettext, autopoint, autoconf and automake are installed:
``$gettext --version``
``gettext (GNU gettext-runtime) 0.19.8.1``

``$autopoint --version``
``/usr/bin/autopoint (GNU gettext-tools) 0.19.8.1``

``$autoconf --version``
``autoconf (GNU Autoconf) 2.69``

``$automake --version``
``automake (GNU automake) 1.16.1``

I'm pretty stumped at this point.. any other info I could provide you that might help?

**Comment by StefanSchippers** [comment]
Can you please try to build gaw3-20200922.tar.gz, get it from here:
https://www.rvq.fr/php/ndl.php?id=gaw.?-.*

This is the original gaw version i had forked to add xschem specific stuff. If this works, i will revert the autoconfig files to original.

**Comment by StefanSchippers** [comment]
Anthony, i have reverted some autoconf files to the original gaw version i forked, and on my system it works fine. 
Please try to update xschem-gaw, do a 'make distclean' to clean up and then run ./configure again.

**Comment by anthonyjwall** [comment]
I've tried both reverting to gaw3-20200922 and updating xschem-gaw and no joy - same error:
``./configure: line 5347: syntax error near unexpected token `external'``
``./configure: line 5347: `AM_GNU_GETTEXT(external)'``

since it works on your system I assume there's some missing dependency or environment variable.

**Comment by StefanSchippers** [comment]
Please try again cloning xschem-gaw, do not do aclocal/automake etc, just run configure.
```
cd xschem-gaw/
./configure
make
```

**Comment by anthonyjwall** [comment]
Removing the repo, cloning again, just running `./configure`, I get a makefile not found error:
``$./configure``
``etc. etc.``
``checking for gtk+-3.0 >= 3.0.0... yes``
``checking for glib-2.0 >= 2.28.0... yes``
``checking that generated files are newer than configure... done``
``configure: creating ./config.status``
``config.status: error: cannot find input file: `Makefile.in'``

`make` then fails of course since no makefile is found:
``$ make``
``make: *** No targets specified and no makefile found.  Stop.``

**Comment by StefanSchippers** [comment]
Please look at the config.log file for pooblems processing Makefile.in.

I can not help much, i only know sutotools generates more problems than the
ones it tries to solve for building applications.
For this readon xschem does no use sutotools (although it has a configure
system). For gaw however getting rid od autotools requires considerable
effort as i am not the original developer. If the problems you face happens
also on the original gaw3-202*****.tar.gz you can try to contact the
original author.

Stefan


Il Ven 24 Giu 2022, 12:45 Anthony Wall ***@***.***> ha
scritto:

**Comment by anthonyjwall** [comment]
Unfortunately `config.log` isn't giving much info either... Just mentions the Makefile in one place:
```
2571 ## ---------------------- ##
2572 ## Running config.status. ##
2573 ## ---------------------- ##
2574 
2575 This file was extended by gaw3 config.status 1.0, which was
2576 generated by GNU Autoconf 2.69.  Invocation command line was
2577 
2578   CONFIG_FILES    =
2579   CONFIG_HEADERS  =
2580   CONFIG_LINKS    =
2581   CONFIG_COMMANDS =
2582   $ ./config.status
2583 
2584 on workstation
2585 
2586 config.status:934: error: cannot find input file: `Makefile.in'
```


Interestingly, all versions of `gaw3` that I've tried compile and install just fine for me. 

As an aside, I've tried upgrading one of my machines to Ubuntu 22.04 so I have gettext 0.21, but have the same error. 

I'll keep playing around... I'll let you know if I come up with any solution.

**Comment by StefanSchippers** [comment]
Thank you for testing.
xschem-gaw is just a small change  over gaw3, so i think i need to revert
all autotools files to the ones of gaw3, update only the .po files (links
to internationalized texts) and commit.
will do in next days.


Il Sab 25 Giu 2022, 13:11 Anthony Wall ***@***.***> ha
scritto:

**Comment by anthonyjwall** [comment]
Ok I found the problem... 
My `gettext` was from `libglib2.0-dev`, so the installed library referenced `AM_GLIB_GNU_GETTEXT_*` instead of `AM_GNU_GETTEXT_*`

So I could call `gettext --version` and get a response - gettext was indeed installed, but the libraries were different. 

I ran `sudo apt install gettext` (different to previously ran `sudo apt install libglib2.0-dev`), re-cloned and the install ran correctly.

Sincere apologies for dragging you along with this one. Hopefully this might help someone in the future with the same issue.
